kitsune/wiki/tests/test_templates.py

```diff
@@ -192,20 +192,36 @@ def test_document_fallback_with_translation_english_slug(self):
 
     def test_document_fallback_no_translation(self):
         """The document template falls back to English if no translation
-        exists."""
+        exists. The fallback message is only shown when there is any revision
+        ready for localization"""
         r = revision(save=True, content='Some text.', is_approved=True)
         url = reverse('wiki.document', args=[r.document.slug], locale='fr')
         response = self.client.get(url)
         doc = pq(response.content)
         eq_(r.document.title, doc('article h1.title').text())
 
+        # Fallback message is not shown because the revision is not ready for
+        # localization.
+        eq_(0, len(doc('#doc-pending-fallback')))
+        # Included content is English.
+        eq_(pq(r.document.html)('div').text(), doc('#doc-content div').text())
+
+        # Marking the document and the revision ready for localization
+        r2 = revision(save=True, document=r.document, content='Some text.',
+                      is_approved=True, is_ready_for_localization=True)
+
+        url = reverse('wiki.document', args=[r.document.slug], locale='de')
+        response = self.client.get(url)
+        doc = pq(response.content)
+
+        eq_(r2.document.title, doc('article h1.title').text())
         # Fallback message is shown.
         eq_(1, len(doc('#doc-pending-fallback')))
         # Removing this as it shows up in text(), and we don't want to depend
         # on its localization.
         doc('#doc-pending-fallback').remove()
         # Included content is English.
-        eq_(pq(r.document.html)('div').text(), doc('#doc-content div').text())
+        eq_(pq(r2.document.html)('div').text(), doc('#doc-content div').text())
```

Comment:
All of this really should be a new test. This test previously tested one thing: document fell back when they don't have translations. Now it it is testing two things: 1) that it falls back, and 2) if the message shows in certain situations. This should be two tests.